validator/sawtooth_validator/execution/processor_iterator.py

```diff
@@ -144,6 +156,34 @@ def cancellable_wait(self, processor_type, cancelled_event):
                 lambda: processor_type in self or cancelled_event.is_set()
             )
 
+    def wait_for_vacancy(self, processor_type, cancelled_event):
+        """Waits for a particular processor type to have the capacity to
+        handle additional transactions or until is_cancelled is True.
+
+        Args:
+            processor_type (ProcessorType): The family, and version of
+                the transaction processor.
+            cancelled_event (threading.Event): is_set() will return True when
+                the wait is cancelled.
+
+        Returns:
+            None
+        """
+
+        with self._condition:
+            self._condition.wait_for(lambda: (
+                self.processor_available(processor_type)
+                or cancelled_event.is_set()))
+
+    def processor_available(self, processor_type):
+        with self._condition:
```

Comment:
I do not think you need to reacquire this condition variable here. I would make this a protected function (start it with an underscore) to show that it should only be used internally.